### ISSUE TITLE
feat: add a delta G field for reactions/metabolites

### ISSUE THREAD

**Issue by feiranl** [SysBioChalmers/RAVEN#280]
### Description of the issue:
* It would be great if delta G field can be added for each reaction/metabolite
* Metacyc offers this information  at pH 7.3 and ionic strength 0.25.

**I hereby confirm that I have:**
- [x] Followed the [guidelines](https://github.com/SysBioChalmers/RAVEN/wiki/Installation) to install RAVEN.
- [x] Checked that a similar issue does not [already exist](https://github.com/SysBioChalmers/RAVEN/issues?utf8=%E2%9C%93&q=is%3Aissue)
- [x] If suitable, needed, asked first in the [Gitter chat room](https://gitter.im/SysBioChalmers/RAVEN) about the issue

**Comment by Hao-Chalmers** [comment]
@feiranl thanks for posting the issue.

This could be a useful feature for GEMs. It can be added as a reaction-associated feature, according to the MetaCyc scheme.

**Comment by simas232** [comment]
Nice idea! Should we consider using delta G values to determine reaction reversibility? At least, it would be nice to have such function in RAVEN beyond MetaCyc reconstruction module

**Comment by Hao-Chalmers** [comment]
@simas232 if SEED applies this value for determining reversibility?

**Comment by simas232** [comment]
Not sure about SEED, but I recall that GEMs with reaction reversibility info determined from delta G had feasibility problems. Let's think and discuss about this functionality over weeks before doing an actual implementation.

I hereby fully support delta G for rxns inclusion in RAVEN model structure. If we include the corresponding info for mets, we need to think how to store this. Probably in similar way like S matrix.

**Comment by edkerk** [comment]
Interesting proposal, should this also stored in SBML when the model is exported? If so, in which field? As far as I know, SBML L3V1 FBCv2 does not have a field for such information.

**Comment by simas232** [comment]
Reaction specific delta G values can be stored in `modelSBML.reaction.notes` while using e.g. "deltaG:" identifier string

**Comment by edkerk** [comment]
Which deltaG value should be stored? @feiranl mentions that MetaCyc provides this information at pH 7.3 and ionic strength 0.25. But the standard Gibbs free energy (ΔG°') is normally measured at pH 7. Probably not a major difference, but it would be good to define a specific condition and provide Gibbs free energies for those.

**Comment by Hao-Chalmers** [comment]
@edkerk MetaCyc does provide deltaG only at unique condition (pH 7.3 and ionic strength 0.25). What other conditions do you suggest?

**Comment by edkerk** [comment]
It is important to have a clear definition of the condition, as changing any parameter will change the Gibbs free energy. Standard Gibbs free energy (ΔG°') is defined at pH 7, 298 K and 1 M concentration of reactants. MetaCyc's condition at pH 7.3 is different, but probably close enough to not result in drastically different values.

What temperature is defined for MetaCyc's condition? It is important to define the condition precisely, because if a reaction is not in MetaCyc then the Gibbs free energy could be calculated by alternative approaches but it should then be for the same condition, so we don't end up with a model filled with Gibbs free energies from very different conditions.

While I would personally prefer standard Gibbs free energy (ΔG°') at pH 7, 298 K and 1 M, as this is even mentioned in biology textbooks, it is probably more convenient to use MetaCyc's condition (and it will not be so much different anyway).

tl;dr: Precisely define MetaCyc's condition and use the same condition for other methods to estimate deltaG values.

**Comment by simas232** [comment]
@Hao-Chalmers, there must be a basis why MetaCyc curators decided to include delta G values upon pH 7.3, not pH 7. Could you please write a quick question to them and ask about this? You may thereby ask about the availability of additional details for conditions these delta G values were measured upon (concentration of reactants and products, temperature, pressure).

**Comment by edkerk** [comment]
To support the work on yeast-GEM: https://github.com/SysBioChalmers/yeast-GEM/pull/330.

The `metDeltaG` and `rxnDeltaG` fields are `double` vectors with same length as `mets` and `rxns`.

- [ ] Add support by `writeYAMLmodel` and `readYAMLmodel` for handling the above fields.
- [ ] Mention the fields in the field definition on the Wiki.
- [ ] Add support of any other model-modifying function to handle the above fields.
- [ ] ***Discussion point:*** Consider if there is a way that the above fields could be considered in SBML file, by `importModel` and `exportModel`.
